recipes/hwloc/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+        "fPIC": [True, False],
+        "with_libxml2": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libxml2": False
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        if self.options.with_libxml2:
+            self.requires("libxml2/2.9.12")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        if self.settings.os == "Windows":
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            deps = CMakeDeps(self)
+            deps.generate()
+            tc = CMakeToolchain(self)
+            tc.cache_variables["HWLOC_ENABLE_TESTING"] = 'OFF'
+            tc.cache_variables["HWLOC_SKIP_LSTOPO"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_TOOLS"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_INCLUDES"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_OPENCL"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_CUDA"] = 'OFF'
+            tc.cache_variables["HWLOC_BUILD_SHARED_LIBS"] = self.options.shared
+            tc.cache_variables["HWLOC_WITH_LIBXML2"] = self.options.with_libxml2
+            tc.generate()
+        else:
+            deps = PkgConfigDeps(self)
+            deps.generate()
+            tc = AutotoolsToolchain(self)
+            if not self.options.with_libxml2:
+                tc.configure_args.extend(["--disable-libxml2"])
+            tc.configure_args.extend(["--disable-io", "--disable-cairo"])
+            tc.generate()
+
+    def build(self):
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.configure(build_script_folder=os.path.join("contrib", "windows-cmake"))
+            cmake.build()
+        else:
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
+
+    def package(self):
+        copy(self, "COPYING", self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.install()
+            # remove PDB files
+            rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        else:
+            autotools = Autotools(self)
+            autotools.install()
+            fix_apple_shared_install_name(self)
+            # remove tools
+            rmdir(self, os.path.join(self.package_folder, "bin"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
```

Comment:
which files are typically stored in this location by this package, and why are they not needed?

Reply:
```
% tree -L 2 share
share
├── applications
│   └── lstopo.desktop
├── bash-completion
│   └── completions
├── doc
│   └── hwloc
├── hwloc
│   ├── hwloc-ps.www
│   ├── hwloc-valgrind.supp
│   ├── hwloc.dtd
│   ├── hwloc2-diff.dtd
│   └── hwloc2.dtd
└── man
    ├── man1
    ├── man3
    └── man7

12 directories, 5 files
```